Add support for dot notation get/set with flag

natanavra opened the pull request:
Settings support key dot notation escape, e.g. `editor.set('city\\.name', 'mycity');`
But get doesn't work with the same `editor.get('city\\.name'); // undefined`

The proposed solution is to simply add a flag for readability:
```js
editor.set('city.name', { ignoreDots: true });
editor.get('city.name', { ignoreDots: true }); // mycity
```

Added the `ignoreDots` option to `set` to avoid hard to read code (non matching keys):
```js
// would not work
const key = 'city\\.name';
editor.set(key, 'mycity');
editor.get(key, { ignoreDots: true }) // undefined
```

IonicaBizau commented:
Maybe it would be better to implement this at instance level? Great feature btw!

natanavra commented:
@IonicaBizau Done, moved `ignoreDots` to the constructor and added a description in the README.
Please be sure to fix the whitespace, for some reason I couldn't get it working in my editor (tabs/spaces ¯\_(ツ)_/¯)

IonicaBizau commented on lib/index.js:
```diff
@@ -29,6 +29,7 @@ class JsonEditor {
         options.stringify_width = options.stringify_width || 2
         options.stringify_fn = options.stringify_fn || null
         options.stringify_eol = options.stringify_eol || false
+	options.ignoreDots = options.ignoreDots || false;
```
Can you please use underscore notation, to be consistent with the other options? Thanks!

bmish commented:
I'm adding a TypeScript definition for this option here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56448
